code/game/machinery/kitchen/microwave.dm

```diff
@@ -308,9 +311,20 @@
 		stop()
 		if(cooked)
 			cooked.loc = src.loc
+			convert_nutriment(cooked)
 		return
 
-/obj/machinery/microwave/proc/wzhzhzh(var/seconds as num) // Whoever named this proc is fucking literally Satan. ~ Z
+/obj/machinery/microwave/proc/convert_nutriment(atom/target)
+	if(blacklisted_from_buffs.Find(target.type))
+		return FALSE // No buffs for you , lazy cook.
+	var/amount = target.reagents.get_reagent_amount("nutriment")
+	target.reagents.remove_reagent("nutriment", amount)
+	target.reagents.add_reagent("cooked_nutriment", amount)
+	amount = target.reagents.get_reagent_amount("protein")
+	target.reagents.remove_reagent("protein", amount)
+	target.reagents.add_reagent("cooked_protein", amount)
```

Comment:
What about other subtypes of nutriment?

Reply:
The other subtypes don't matter at all , the main nutrients that food has is animal protein and nutriment. the others are more for flavoring and do very little in terms of nutrition